src/backend/env/mod.rs

```diff
@@ -1671,18 +1683,26 @@ impl State {
             }
         };
 
-        if last_weekly_chores + WEEK < now {
+        if timers.last_weekly + WEEK < now && !timers.weekly_pending {
+            mutate(|state| {
+                state.timers.weekly_pending = true;
```

Comment:
If the canister traps during execution of the chore, then it will never retry the chore again due to the pending flag. Is the plan for such cases to reset the flags with an upgrade?

Reply:
This is correct. I remember I've read somewhere that there is a way to do some clean ups on panics too, but I couldn't find [anything useful](https://github.com/dfinity/cdk-rs/blob/7fa27b03d0d891b30a89b1b281ca3ba0661f2750/src/ic-cdk/src/api/call.rs#L114) in the rust CDK. So, currently I prefer a manual fix via upgrade, but mostly because I do not know yet how to do better... Any advice?

Reply:
There is a "guard" pattern that uses the `drop` destructor to cleanup code after panics. See for example: https://github.com/dfinity/ic/blob/9cad8170f98b8fd58f392b2395adf0a7f1d3e3c2/rs/ethereum/cketh/minter/src/guard/mod.rs